File: Backend/schemas/jira.py
from pydantic import BaseModel

# 前端顯示用格式（GET /jira/issues 的 response_model）——crud/jira.py 的
# transform_jira_item() 已經把 Jira 原始 API 那種多層巢狀結構（fields.status.name、
# fields.assignee.displayName 等）拉平成單層
class JiraIssue(BaseModel):
    id: str
    key: str
    summary: str
    status: str
    updated: str
    assignee: str
    avatar: str
    type: str
    iconUrl: str

File: Frontend/types/jira.ts
// 對應後端 schemas/jira.py 的 JiraIssue——後端已經把 Jira 原始 API 的巢狀結構
// 拉平成單層，這裡每個欄位後端都保證會給字串（抓不到就是空字串），不是 optional
export interface JiraIssue {
    id: string
    key: string
    summary: string
    status: string
    updated: string
    assignee: string
    avatar: string
    type: string
    iconUrl: string
  }